source/priority_queue.c

```diff
@@ -91,43 +145,123 @@ void aws_priority_queue_init_static(
     aws_priority_queue_compare_fn *pred) {
 
     queue->pred = pred;
+    memset(&queue->backpointers, 0, sizeof(queue->backpointers));
+
     aws_array_list_init_static(&queue->container, heap, item_count, item_size);
 }
 
 void aws_priority_queue_clean_up(struct aws_priority_queue *queue) {
     aws_array_list_clean_up(&queue->container);
+    aws_array_list_clean_up(&queue->backpointers);
 }
 
 int aws_priority_queue_push(struct aws_priority_queue *queue, void *item) {
+    return aws_priority_queue_push_ref(queue, item, NULL);
+}
+
+int aws_priority_queue_push_ref(
+    struct aws_priority_queue *queue,
+    void *item,
+    struct aws_priority_queue_node *backpointer) {
     int err = aws_array_list_push_back(&queue->container, item);
     if (err) {
         return err;
     }
 
+    size_t index = aws_array_list_length(&queue->container) - 1;
+
+    if (backpointer && !queue->backpointers.alloc) {
+        if (!queue->backpointers.alloc) {
```

Comment:
why is this branch here when we already know this condition has to be true in order to get here?

Reply:
oops, leftover from a code reorganization